### PR TITLE
Exposing anonymized method for logging 

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue #766.
New **ParseResult.GetAnonymizedFormula()** method is public but interfaced used to inject behavior to **StructuralPrint** aren't:
- **TexlBinding**
- **ISanitizedNameProvider**

What's the best approach?

Update: checked internally. Injection classes won't be exposed (for now).

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Parser/ParseResult.cs [hunk 1]
```diff
@@ -72,5 +73,13 @@ internal ParseResult(TexlNode root, List<TexlError> errors, bool hasError, List<
             err.FormatCore(sb);            
             return $"Err#{++i} {sb}";
         }));
+
+        /// <summary>
+        /// Converts the current formula into an anonymized format.
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Parser/ParseResult.cs:78]: >format [](http://example.com/codeflow?start=60&length=6)

"... suitable for logging"

File: src/libraries/Microsoft.PowerFx.Core/Parser/ParseResult.cs [hunk 2]
```diff
@@ -72,5 +73,13 @@ internal ParseResult(TexlNode root, List<TexlError> errors, bool hasError, List<
...
+        /// </summary>
+        public string GetAnonymizedFormula()
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Parser/ParseResult.cs:80]: >GetAnonymizedFormula [](http://example.com/codeflow?start=22&length=20)

This is good. 
And as a method, we can add overloads with other parameters later if we want to customize.

File: src/tests/Microsoft.PowerFx.Core.Tests/FormatterTests.cs [hunk 3]
```diff
@@ -35,7 +35,8 @@ public void TestStucturalPrint(string script, string expected)
                 _defaultLocale,
                 flags: Flags.EnableExpressionChaining);
 
-            Assert.Equal(expected, StructuralPrint.Print(result.Root));
+            //Assert.Equal(expected, StructuralPrint.Print(result.Root));
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/FormatterTests.cs:38]: > //Assert.Equal(expected, StructuralPrint.Print(result.Root)); [](http://example.com/codeflow?start=11&length=62)

nit, dead code

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**